app/controllers/miq_policy_controller/alert_profiles.rb

```diff
@@ -343,4 +347,28 @@ def alert_profile_get_info(alert_profile)
     @right_cell_text = _("%{model} \"%{name}\"") % {:model => ui_lookup(:model => "MiqAlertSet"), :name => alert_profile.description}
     @right_cell_div = "alert_profile_details"
   end
+
+  def alert_profile_sync_provider(old_alerts = nil, new_alerts = nil)
+    if @alert_profile.mode == "MiddlewareServer"
+      if old_alerts.nil? && new_alerts.nil?
+        operation = :update_assignments
+        old_alerts = new_alerts = @alert_profile.miq_alerts.collect(&:id)
+        assigned = @alert_profile.get_assigned_tos
```

Comment:
what `_tos` stands for? In the impl, I can see "Returns: {:objects => [obj, obj, ...], :tags => [[Classification.entry_object, klass], ...]}"

Do I get it right, that it retrieves the tags (the MiQ ones) and tries to update the alert definition in the Hawkular services with those tags (+ other metadata)?